src/GraphQL/DI/IConfigureExecution.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+
+namespace GraphQL.DI
```

Comment:
There is nothing specific to DI 😉 . Even `DefaultServiceProvider` and `FuncServiceProvider` are placed in `GraphQL` namespace. `IGraphQLBuilder`/`ServiceLifetime` indeed have DI-specific intentions but... I don't know how to be in that situation. We can go another way and change namespace for `DefaultServiceProvider` and `FuncServiceProvider` for `GraphQL.DI` in v5.

Reply:
> We can go another way and change namespace for `DefaultServiceProvider` and `FuncServiceProvider` for GraphQL.DI in v5.

Sounds good to me.

I'm open to discussions about what namespace to put this stuff in.  Just seems to make sense to me to have the DI-related classes and interfaces together.

I did put `GraphQLBuilderExtensions` in the root `GraphQL` namespace.  This is so that users only need `using GraphQL;` to be able to use all of the extension methods.  However, the other nuget packages are in their own namespaces, so you have to `using GraphQL.SystemTextJson;` for example.  This makes it clear and eliminates ambiguity by the compiler of what extension method you might be attempting to use.  Finally, the `AddGraphQL()` extension method in `GraphQL.MicrosoftDI` also requires `using GraphQL.MicrosoftDI`.  I felt it was wrong of the server project to place the extension method directly in the Microsoft namespace, even though this made it so that the using statement was not required.  I looked at some other MS implementations and felt that putting the method in `GraphQL.MicrosoftDI` was correct.

I'm open to discussions on all of that as well.

Reply:
I follow the rule that the extension method for type T should be located in the same namespace as T.

>  This makes it clear and eliminates ambiguity by the compiler of what extension method you might be attempting to use.

No ambiguity/collisions if you choose good names.

Reply:
> I follow the rule that the extension method for type T should be located in the same namespace as T.

Ok; that is not how I decided to set it up.  We can discuss changing this in a separate PR/issue.

Reply:
I added TODO in code.